third_party/2and3/sqlalchemy/engine/base.pyi

```diff
@@ -1,20 +1,28 @@
-from typing import Any, List, Tuple
+from typing import Any, Iterable, Iterator, List, Mapping, Optional, Tuple
 
 # Dummy until I figure out something better.
 class Connectable:
-    pass
+    def execute(self, object, *multiparams, **params) -> ResultProxy: ...
 
-class Connection:
+class Connection(Connectable):
     def begin(self): ...
-    def execute(self, object, *multiparams, **params): ...
 
-class Engine(object): ...
+class Engine(Connectable): ...
+
+class RowProxy(Mapping[str, Any]):
+    pass
 
-class RowProxy:
-    def items(self) -> List[Tuple[Any, Any]]: ...
-    def keys(self) -> List[Any]: ...
-    def values(self) -> List[Any]: ...
-    def __getitem__(self, key: str): ...
+class ResultProxy(Iterator[RowProxy]):
+    def keys(self) -> List[str]: ...
+    def close(self) -> None: ...
+    def __iter__(self) -> Iterator[RowProxy]: ...
+    def fetchall(self) -> Iterator[RowProxy]: ...
+    def fetchmany(self, size: int=None) -> Iterator[RowProxy]: ...
+    def fetchone(self) -> Optional[RowProxy]: ...
+    def first(self) -> Optional[RowProxy]: ...
+    def scalar(self): ...
```

Comment:
Very likely an union of [DB API Type Objects](https://www.python.org/dev/peps/pep-0249/#type-objects-and-constructors). By brief search of this repo they don't seem to be defined anywhere to be shared except as internal part of sqlite and mysql.

Would it be ok to use `Any` with a comment reference to future union type?

This is also related to the `RowProxy` value type which should be of this union.